core/src/main/java/io/aiven/klaw/config/ManageDatabase.java

```diff
@@ -98,7 +99,7 @@ public class ManageDatabase implements ApplicationContextAware, InitializingBean
 
   // key tenantId, sub key clusterid Pertenant
   private static Map<Integer, Map<Integer, KwClusters>> kwKafkaConnectClustersPertenant;
-
+  private static Map<Integer, Map<String, Env>> envMapPerTenant = new HashMap<>();
```

Comment:
there are few existing maps like allEnvListPerTenant can be reused ?

Reply:
allEnvListPerTenant means we have to iterate through the list every time to find the environment we are looking for.

The map for this particular use case is a much less expensive operation and is something that we could use in a lot more places imo.